### DISCUSSION TITLE
Mender install on Jetson TX2i failed

### DISCUSSION THREAD

**Discussion by ghost** [OE4T/tegra-demo-distro#172] [Q&A] [answered]
Hi 
I am trying to update demo image with mender artifact for devkit with Jetson TX2 and Jetson TX2i.
I cloned the **tegra-demo-distro** (branch **dunfell-l4t-r32.5.0**) and tried to build **demo-image-egl** (**DISTRO = "tegrademo-mender"**) and to update it using Mender. 
I haven't done any modifications of distro, image, machines. 
After each successfull build, I flashed the module, boot it, after a successful boot, I copied **demo-image-egl-jetson-tx2-devkit-tx2i.mender** (or **demo-image-egl-jetson-tx2-devkit.mender**) to the **/home/root/** and launched **mender install demo-image-egl-jetson-tx2-devkit(-tx2i).mender**. Below described configurations and results:

1. **MACHINE = jetson-tx2-devkit** - mender install successfull

2. **MACHINE = jetson-tx2-devkit-tx2i** - mender install failed with log:
```
root@jetson-tx2-devkit-tx2i:~# mender install demo-image-egl-jetson-tx2-devkit-tx2i.mender 
INFO[0000] Loaded configuration file: /var/lib/mender/mender.conf 
INFO[0000] Loaded configuration file: /etc/mender/mender.conf 
INFO[0000] Mender running on partition: /dev/mmcblk0p1  
INFO[0000] Start updating from local image file: [demo-image-egl-jetson-tx2-devkit-tx2i.mender] 
Installing Artifact of size 281420800...
INFO[0000] No public key was provided for authenticating the artifact 
INFO[0000] Update Module path "/usr/share/mender/modules/v3" could not be opened (open /usr/share/mender/modules/v3: no such file or directory). Update modules will not be available 
INFO[0000] Opening device "/dev/mmcblk0p33" for writing 
INFO[0000] Native sector size of block device /dev/mmcblk0p33 is 512 bytes. Mender will write in chunks of 1048576 bytes 
.............................................................. - 100 %
INFO[0074] All bytes were successfully written to the new partition 
INFO[0074] The optimized block-device writer wrote a total of 11772 frames, where 433 frames did need to be rewritten (i.e., skipped) 
INFO[0076] Wrote 12343259136/12343259136 bytes to the inactive partition 
INFO[0076] Enabling partition with new image installed to be a boot candidate: 33 
INFO[0076] Executing script: ArtifactInstall_Leave_80_bl-update 
INFO[0076] Collected output (stderr) while running script /var/lib/mender/scripts/ArtifactInstall_Leave_80_bl-update
Error: missing entries for partitions: bpmp-fw-dtb, bootloader-dtb, VER, BCT, MB1_BCT, kernel-dtb
       for TNSPEC 3489-300-0000-S.0-1-0-jetson-tx2-devkit-tx2i-mmcblk0p1
ERR: bootloader update failed

---------- end of script output 
ERRO[0076] ArtifactInstall_Leave script failed: statescript: error executing 'ArtifactInstall_Leave_80_bl-update': 1 : exit status 1 
Rolling back Artifact...
INFO[0076] Rolling back to the active partition: (1).   
ERRO[0076] statescript: error executing 'ArtifactInstall_Leave_80_bl-update': 1 : exit status 1
```
I got similar results when switching **tegra-demo-distro** to branch **hardknott**

Does the mender currently support **jetson-tx2-devkit-tx2i**? Maybe I did something wrong? Please, help.

**Comment by madisongh** [comment, marked as answer]
It's not that Mender specifically is unsupported on the TX2i, but more generally for the TX2i we rely on the community of TX2i users to help with support, as it's one of the Jetson modules I do not have and so cannot test with.

It looks like the problem is in how we're generating the bootloader update payload - the platform-specific entries in the payload are getting tagged with a board SKU of 1000 rather than 0000.  You should be able to work around the problem by adding
```
TEGRA_BOARDSKU_jetson-tx2-devkit-tx2i = "0000"
```
in your `local.conf` file.

**Reply by madisongh** [reply]
I've updated the current branches in tegra-demo-distro with the latest meta-tegra layer which fixes this problem without having to use the workaround I mentioned above. If you update to latest on `dunfell-l4t-r32.5.0` and `git submodule update --init` to pull in the updated meta layers and build yourself a new Mender artifact, it should install OK.

The relevant PR in meta-tegra is [here](https://github.com/OE4T/meta-tegra/pull/851).

**Comment by ghost** [comment]
Many thanks!
I've pulled layers, rebuilt the image, performed **mender install** with **MACHINE = jetson-tx2-devkit-tx2i**. Everything works perfectly!
